x-pack/legacy/plugins/apm/server/lib/helpers/setup_request.ts

```diff
@@ -47,35 +40,70 @@ export interface Setup {
   uiFiltersES: ESFilter[];
   client: ESClient;
   internalClient: ESClient;
-  config: KibanaConfig;
+  config: APMConfig;
   indices: ApmIndicesConfig;
   dynamicIndexPattern?: IIndexPattern;
 }
 
-export async function setupRequest(req: Legacy.Request): Promise<Setup> {
-  const query = (req.query as unknown) as APMRequestQuery;
-  const { server } = req;
-  const savedObjectsClient = server.savedObjects.getScopedSavedObjectsClient(
-    req
-  );
-  const config = server.config();
-  const indices = await getApmIndices({ config, savedObjectsClient });
+export type SetupWithTimeRange = Setup & {
+  start: number;
+  end: number;
+};
+
+export type SetupWithUIFilters = Setup & {
+  uiFiltersES: ESFilter[];
+};
+
+export type SetupWithAllFilters = SetupWithTimeRange & SetupWithUIFilters;
```

Comment:
I'm not sure this'll work longer term. What happens when we add a new property that not every route needs? Then it's not "all" anymore.

Perhaps it's better to be explicit and type it in place as: `Setup & TimeRange & UIFilters`.

Not for this PR but: `Setup` itself is already pretty opaque. It contains a bunch of different stuff - perhaps it would make sense to split that up into smaller parts too?

Reply:
++ on separating Setup{WithTimeRange,WithUIFilters} and using TS unions. Composable & explicit.

I'd think it'd be good to think (at some point) about APMRequestContext and SetupRequest, how they overlap, how they are different, and possibly if we need a APMPluginContract or something similar as well. Maybe after NP migration is completed?